src/typeguards.ts

```diff
@@ -0,0 +1,31 @@
+import type { Entry } from 'contentful'
+import type { DeprecatedExperience, Experience, ExperienceEntry } from './types'
+
+// @ts-expect-error type incompatibility
+export const isExperienceEntry = (entry: Entry): entry is ExperienceEntry => {
```

Comment:
One reason why this breaks TypeScript probably is that we're using the `Entry` type from the CDA library which requires the two methods `toPlainObject()` and `update()` which we probably don't have here if I'm not mistaken. So eventually, this can be fixed by using the correct type for the parameter.
For reference, in the experience builder we have a similar checker in `schemaDetector.ts` where the parameter has the type `Entry` (from cma library not cda) and assesses that it's actually an `ExperienceEntry`.

![Screenshot 2023-10-23 at 12 12 33](https://github.com/contentful/experience-builder/assets/9327071/9fee5009-1105-4392-a710-8a701c124a3a)
![Screenshot 2023-10-23 at 12 15 31](https://github.com/contentful/experience-builder/assets/9327071/ca0032cb-b179-4981-b3cc-48d6f4a431ff)

Reply:
To avoid all these ugly suppression comments in the file, maybe we should just change the type to `any` or (better) `unknown` until we use a proper type for `Entry` in this codebase.

Reply:
Mmmh, looking at the logic in `fetchers` I see that the entry is actually coming from the CDA fetch method and thus correct. So it's the returned type here that is not correct.